### PR TITLE
New exogenous carbon price interface for coupling

### PR DESCRIPTION
**Pull request by gabriel-abrahao**
This PR will implement a new setting in `scenario_config` and another in `scenario_config_coupled` that allows one to set a separate exogenous carbon price path in REMIND and in MAgPIE coupled, coming from another run.

- [x] Exogenous carbon price for REMIND standalone
   - [x] Update `start` 
      - [x] Add `path_gdx_carbonprice` to `files2export`
      - [x] Make sure the existence of  `path_gdx_carbonprice` is checked every time before trying to access it
   - [x] Update `prepare_and_run` 
      - [x] Make `start_subsequent_runs` fill the paths of the previous run
      - [x] Add a condition in the `files2export` loop to export only the relevant variables (no need to copy a 500MB file)
      - [x] Use `create_ExogSameAsPrevious_CO2price_file` to create input file `remind/modules/45_carbonprice/exogenous/input/p45_tau_co2_tax.inc`
   - [x] Rename `create_ExogSameAsPrevious_CO2price_file` to  `create_input_for_45_carbonprice_exogenous`
   - [x] Harmonize format and units the two scripts (`scripts/input/r_gen_co2_tax.R` and `scripts/input/create_input_for_45_carbonprice_exogenous.R`) use for their output file `remind/modules/45_carbonprice/exogenous/input/p45_tau_co2_tax.inc` 
   - [x] Delete the `ExogSameAsPrevious` `45_carbonprice` realization

- [x] Exogenous carbon price for REMIND coupled
   - [x] Update `start_bundle_coupled`
      - [x] Add `path_gdx_carbonprice` to `files2export`
      - [x] Make sure the existence of  `path_gdx_carbonprice` is checked every time before trying to access it

- [x] Exogenous carbon price for MAgPIE
   - [x] Update the `start_bundle_coupled.R` and `start_coupled.R` to export a different price for MAgPIE
   - [x] Update MAgPIE code (in different github repo) to import carbon prices from a different file

- [x] Tests
   - [x] Test standalone: test if REMIND uses an exogenous carbon price
   - [x] Test with an old `scenario_config` (that doesn't have a `path_gdx_carbonprice`  column)
   - [x] Test coupled
      - [x] Test if different carbon price is given to MAgPIE
      - [x] Test if REMIND uses exogenous carbon price also in coupled mode

### CODE REVIEW

**Review by dklein-pik — APPROVED** [review]

File: core/sets.gms [hunk 1]
```diff
@@ -874,7 +874,6 @@ $endif.altFeEmiFac
 ***######################## R SECTION START (MODULES) ###############################
 *** THIS CODE IS CREATED AUTOMATICALLY, DO NOT MODIFY THESE LINES DIRECTLY
 *** ANY DIRECT MODIFICATION WILL BE LOST AFTER NEXT MODEL START
-*** CHANGES CAN BE DONE USING THE RESPECTIVE LINES IN scripts/start_functions.R
```
[the remark this answers is not in the document]
Comment (dklein-pik) [reply core/sets.gms:877]: Because there is no scripts/start_functions.R enymore ;-)

[the remark this answers is not in the document]
Comment (LaviniaBaumstark) [reply core/sets.gms:877]: ah, yes, but are parts of it done in prepare_and_run?

[the remark this answers is not in the document]
Comment (dklein-pik) [reply core/sets.gms:877]: Should be scripts/start/prepare_and_run.R then, right?

[the remark this answers is not in the document]
Comment (LaviniaBaumstark) [reply core/sets.gms:877]: I think so

**Review by LaviniaBaumstark — APPROVED** [review]

**Pull request merged**